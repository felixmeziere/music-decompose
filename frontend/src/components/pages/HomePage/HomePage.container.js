import { connect } from 'react-redux';
import {
  addSong,
  setSongUploadField,
} from 'redux/actions/songUpload';
import HomePage from './HomePage';

const mapStateToProps = ({ status, songUpload }) => ({
  loading: status.loading,
  errorMessage: status.errors.songUpload,
  message: status.message,
  title: songUpload.title,
  songFile: songUpload.file,
  songUUID: songUpload.songUUID,
});

const mapDispatchToProps = {
  addSong,
  setSongUploadField,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(HomePage);
