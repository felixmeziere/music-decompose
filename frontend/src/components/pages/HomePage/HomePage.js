import React from 'react';
import PropTypes from 'prop-types';

const HomePage = props => (
  <div>
    <div>Error Message:</div>
    <div>{props.errorMessage}</div>
    <div>Message:</div>
    <div>{props.message}</div>
    <div>Loading?</div>
    <div>{props.loading ? 'Yes' : 'No'}</div>
    <form
      onSubmit={(e) => {
        e.preventDefault();
        props.uploadSong();
      }}
    >
      <input
        onChange={(e) => {
          e.preventDefault();
          props.setUploadSongField('file', e.target.value);
        }}
        type="text"
      />
      <input type="submit" />
    </form>
    <submit />
  </div>
);

HomePage.propTypes = {
  loading: PropTypes.bool.isRequired,
  errorMessage: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  uploadSong: PropTypes.func.isRequired,
  setUploadSongField: PropTypes.func.isRequired,
};

export default HomePage;
