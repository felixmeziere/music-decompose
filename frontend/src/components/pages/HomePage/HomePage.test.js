import React from 'react';
import renderer from 'react-test-renderer';
import HomePage from './HomePage';

it('renders without crashing', () => {
  const component = renderer.create(<HomePage
    loading={false}
    errorMessage="There is an error."
    message="Totor Panache"
    uploadSong={() => {}}
    setUploadSongField={() => {}}
  />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
