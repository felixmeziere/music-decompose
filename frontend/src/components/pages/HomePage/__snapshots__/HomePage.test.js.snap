// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div>
    Error Message:
  </div>
  <div>
    There is an error.
  </div>
  <div>
    Message:
  </div>
  <div>
    Totor Panache
  </div>
  <div>
    Loading?
  </div>
  <div>
    No
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
    />
    <input
      type="submit"
    />
  </form>
  <submit />
</div>
`;
